common/djangoapps/student/tests/test_enrollment.py

```diff
@@ -218,3 +227,126 @@ def _change_enrollment(self, action, course_id=None, email_opt_in=None):
             params['email_opt_in'] = email_opt_in
 
         return self.client.post(reverse('change_enrollment'), params)
+
+    def _enrollment(self):
+        """ Generate a default course mode and enrollment. """
+        mode = CourseMode(
+            course_id=self.course.id,
+            mode_slug='verified',
+            mode_display_name='Verified',
+        )
+        mode.save()
+
+        return CourseEnrollment(
+            mode=CourseMode.VERIFIED,
+            course_id=self.course.id,
+            user=self.user,
+        )
+
+    def test_refundable_can_refund(self):
+        """ Assert that enrollment is refundable if can_refund is set."""
+        enrollment = self._enrollment()
+        enrollment.can_refund = True
+
+        self.assertTrue(enrollment.refundable())
+
+    def test_refundable_generated_cert(self):
+        """ Assert that enrollment is not refundable if cert has been generated."""
+        with patch('student.models.GeneratedCertificate') as generate_cert:
+            instance = generate_cert.return_value
+            instance.certificate_for_student.return_value = True
+            enrollment = self._enrollment()
+
+            self.assertFalse(enrollment.refundable())
+
+    def test_refundable_after_refund_window(self):
+        """ Assert that enrollment is not refundable if it is after the refund_window_end_date."""
+        with patch('student.models.CourseEnrollment._refund_window_end_date') as window:
+            window.return_value = datetime.now() - timedelta(days=1)
+            enrollment = self._enrollment()
+
+            self.assertFalse(enrollment.refundable())
+
+    def test_refundable_before_refund_window(self):
+        """ Assert that enrollment is refundable if it is before the refund_window_end_date."""
+        with patch('student.models.CourseEnrollment._refund_window_end_date') as window:
+            window.return_value = datetime.now() + timedelta(days=1)
+            enrollment = self._enrollment()
+
+            self.assertTrue(enrollment.refundable())
+
+    def test_refundable_no_verified(self):
+        """ Assert that enrollment is not refundable if there is no verified course mode for the course."""
```

Comment:
Why does the verified mode matter? Shouldn't this be based on whether the user paid?